test/integration/targets/hcloud_network/tasks/main.yml

```diff
@@ -0,0 +1,91 @@
+# Copyright: (c) 2019, Hetzner Cloud GmbH <info@hetzner-cloud.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+---
+- name: test missing ip_range parameter on create Network
+  hcloud_network:
+    name: "{{hcloud_network_name}}"
+  register: result
+  ignore_errors: yes
+- name: verify fail missing ip_range parameter on create Network result
+  assert:
+    that:
+      - result is failed
+      - 'result.msg == "missing required arguments: ip_range"'
+
+- name: test create Network with check mode
+  hcloud_network:
+    name: "{{hcloud_network_name}}"
+    ip_range: "10.0.0.0/8"
+  register: result
+  check_mode: yes
+- name: verify create Network with check mode result
+  assert:
+    that:
+      - result is changed
+
+- name: test create Network
+  hcloud_network:
+    name: "{{hcloud_network_name}}"
+    ip_range: "10.0.0.0/8"
+  register: network
+- name: verify test create Network result
+  assert:
+    that:
+      - network is changed
+      - network.hcloud_network.name == "{{hcloud_network_name}}"
+      - network.hcloud_network.ip_range == "10.0.0.0/8"
+
+- name: test create Network idempotence
+  hcloud_network:
+    name: "{{hcloud_network_name}}"
+    ip_range: "10.0.0.0/8"
+  register: network
+- name: verify test create Network idempotence result
+  assert:
+    that:
+      - network is not changed
+      - network.hcloud_network.name == "{{hcloud_network_name}}"
+      - network.hcloud_network.ip_range == "10.0.0.0/8"
+
+- name: test create Network idempotence
+  hcloud_network:
+    name: "{{hcloud_network_name}}"
+    ip_range: "10.0.0.0/8"
+  register: network
+- name: verify test create network
+  assert:
+    that:
+      - network is not changed
+
+- name: test update Network label
+  hcloud_network:
+    name: "{{hcloud_network_name}}"
+    labels:
+      key: value
+  register: network
+- name: verify test update Network lable
+  assert:
+    that:
+      - network is changed
+      - network.hcloud_network.labels.key == "value"
+
+- name: test update Network label with the same label
```

Comment:
Maybe also add idempotency / check mode tests for this one.

Reply:
this is already the idempotency check